src/main/java/Parser.java

```diff
@@ -0,0 +1,106 @@
+public class Parser {
+    private static final String COMMAND_BYE = "bye";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_MARK = "mark";
+    private static final String COMMAND_UNMARK = "unmark";
+    private static final String COMMAND_TODOS = "todo";
+    private static final String COMMAND_DEADLINES = "deadline";
+    private static final String COMMAND_EVENTS = "event";
+
+    private static boolean isInteger(String toCheck) {
+        try {
+            int intValue = Integer.parseInt(toCheck);
+            return true;
+        } catch (NumberFormatException e) {
+            return false;
+        }
+    }
+
+    public static boolean isOnline(String command, TaskManager taskManager) {
+        String keyword = extractKeyword(command);
+        switch(keyword) {
+        case COMMAND_BYE:
+            System.out.println("Bye and see you again soon!");
+            return false;
+        case COMMAND_LIST:
+            taskManager.listTasks();
+            break;
+        case COMMAND_TODOS:
+            processNewTodo(command, taskManager);
+            break;
+        case COMMAND_DEADLINES:
+            processNewDeadline(command, taskManager);
+            break;
+        case COMMAND_EVENTS:
+            processNewEvent(command, taskManager);
+            break;
+        case COMMAND_MARK:
+            processMark(command, taskManager);
+            break;
+        case COMMAND_UNMARK:
+            processUnmark(command, taskManager);
+            break;
+        default:
+            System.out.println("Invalid command");
+            break;
+        }
+
+        return true;
+    }
+
+    private static void processUnmark(String command, TaskManager taskManager) {
+        String[] arrOfCommand = command.split(" ");
+        if (arrOfCommand.length > 1 && isInteger(arrOfCommand[1])) {
```

Comment:
Since there is repeated code here, you could consider restructuring this code such that it abstracts the pattern out

Reply:
*in this file, not this specific line